File: golang/internal/application/repository/portfolio_repository.go
package repository

import "folio/codinginterview/internal/domain"

// PortfolioRepository は最適ポートフォリオのリポジトリインターフェースです。
type PortfolioRepository interface {
	Get() (domain.Portfolio, error)
	Update(portfolio domain.Portfolio) error
}

File: java8/src/main/java/folio/codinginterview/application/repository/PortfolioRepository.java
package folio.codinginterview.application.repository;

import folio.codinginterview.domain.Portfolio;

import java.util.concurrent.CompletableFuture;

/**
 * 最適ポートフォリオリポジトリ。
 */
public interface PortfolioRepository {
    CompletableFuture<Portfolio> get();

    CompletableFuture<Void> update(Portfolio portfolio);
}
